src/sasquatchbackpack/sasquatch.py

```diff
@@ -216,30 +300,95 @@ def _get_source_records(self) -> list[dict] | None:
             if self.redis.get(self.source.get_redis_key(record)) is None
         ]
 
-    def post(self) -> tuple[str, list]:
+    def _check_success(self, response: dict) -> None:
+        """Check whether the request has suceeded.
+
+        Parameters
+        ----------
+        response: dict
+            response object from post().
+        """
+        for item in response["requests"]:
+            if response["requests"][item]["status"] == "Error":
+                response["succeeded"] = False
+                break
+            response["succeeded"] = True
+
+    def post(self, *, force_post: bool = False) -> str:
         """Assemble schema and payload from the given source, then
         makes a POST request to kafka.
 
+        Parameters
+        ----------
+        force_post: bool
+            Ignore checks to determine whether a topic exists and
+            push a new topic name, overwriting any extant topic names.
+
         Returns
         -------
-        response-text : str
-            The results of the POST request in string format
-        records : list
-            List of earthquakes with those already stored on remote removed
+        response: str
+            stringified json containing the following key value pairs:
+            "succeeded": bool - True if the post succeeded,
+            False if any relevant steps failed.
+            "records": list - List of posted records.
+            "requests": dict[str, dict[str, str]] The results of various
+            POST requests sent by this function. Empty values denote a POST
+            request that has not occurred.The keys withn this dictionary are:
+            "check_topic", create_topic", "write_values": dict[str, str].
+            Each contains the following values: status": str - "Success"
+            if request suceeded, "Error" if request failed, "Warning"
+            for other non-breaking behavior. "message": str - Specific
+            description of what occured.
         """
+        response: dict = {
+            "succeeded": False,
+            "requests": {
+                "check_topic": {},
+                "create_topic": {},
+                "write_values": {},
+            },
+            "records": [],
+        }
+        reqs = response["requests"]
+
+        topic_exists = self._topic_exists(reqs)
+
+        if topic_exists is None:
+            return json.dumps(response)
+
+        if force_post or not topic_exists:
+            reqs["create_topic"] = self._post_new_topic()
+            if reqs["create_topic"]["status"] == "Error":
+                return json.dumps(response)
+        else:
+            reqs["create_topic"] = {
+                "status": "Success",
+                "message": (
+                    "Relevant topic already exists in kafka, "
+                    "bypassing creation POST request."
+                ),
+            }
+
```

Comment:
Right I also think that all this logic can be simplified.
You should assume the normal case here and handle exceptions if they occur.  
In other words, we should avoid creating the Kafka topic every you want to send a message to it (and thus also checking for its existence). 
Those are unnecessary requests to the REST Proxy API.
A better solution to this PR is dealing with the Kafka topic creation during "initialization" i.e. by defining a `KafkaTopic` resource and let Strimzi, the cluster operator, to create the topic for your app.
I can show you this approach tomorrow in our tag up.


Let's talk more about this approach in out next tag up.